vck/src/commonMain/kotlin/at/asitplus/wallet/lib/agent/KeyWithCert.kt

```diff
@@ -0,0 +1,89 @@
+package at.asitplus.wallet.lib.agent
+
+import at.asitplus.signum.indispensable.*
+import at.asitplus.signum.indispensable.josef.JsonWebKey
+import at.asitplus.signum.indispensable.josef.toJsonWebKey
+import at.asitplus.signum.indispensable.pki.X509Certificate
+import at.asitplus.signum.indispensable.pki.X509CertificateExtension
+import at.asitplus.signum.supreme.asKmmResult
+import at.asitplus.signum.supreme.sign.EphemeralKey
+import at.asitplus.signum.supreme.sign.Signer
+import io.github.aakira.napier.Napier
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+/**
+ * Abstracts the management of key material away from [CryptoService].
+ */
+interface KeyWithCert : Signer {
```

Comment:
But it's not always with a cert, is it?

Reply:
i am all for a better name!

Reply:
`KeyMaterial` or something like that? Previously, it has been `KeyPairAdapter`

Reply:
done

Reply:
done